src/lib/keccak.ts

```diff
@@ -64,14 +102,156 @@ const ROUND_CONSTANTS = [
   0x8000000000008080n,
   0x0000000080000001n,
   0x8000000080008008n,
-].map(Field.from);
+];
+
+// AUXILARY FUNCTIONS
+
+// Auxiliary function to check composition of 8 bytes into a 64-bit word
+function checkBytesToWord(word: Field, wordBytes: Field[]): void {
+  let composition = wordBytes.reduce((acc, x, i) => {
+    const shift = Field.from(2n ** BigInt(8 * i));
+    return acc.add(x.mul(shift));
+  }, Field.from(0));
+
+  word.assertEquals(composition);
```

Comment:
Also, is this just a normal assertion or it actually produces constraints? If so, I would write this clearly upfront in the docs of the function.

Reply:
I added a comment which explains that this function creates constraints (`assertEquals` creates a constraint).